### DISCUSSION TITLE
How to use gatsby-transformer-javascript-frontmatter correctly with Gatsby 4

### DISCUSSION THREAD

**Discussion by cameronmcefee** [gatsbyjs/gatsby#34167] [Help]
Howdy 🤠 ,

We're using [`gatsby-transformer-javascript-frontmatter`](https://www.gatsbyjs.com/plugins/gatsby-transformer-javascript-frontmatter/) which requires exporting a `frontmatter` object from the javascript files in question. We've got a bunch of pages that export `frontmatter`, but now that I've upgraded to Gatsby 4 (I think this was a thing in 3 as well) I get all kinds of warnings because of these exports:

```
/Users/cameronmcefee/github/static-sites/packages/www/src/pages/terms.js
  6:1  warning  In page templates only a default export of a valid React component and the named exports of a page query, getServerData or config are allowed.
        All other named exports will cause Fast Refresh to not preserve local component state and do a full refresh.

        Please move your other named exports to another file. Also make sure that you only export page queries that use the "graphql" tag from "gatsby".
```

The plugin docs don't mention anything about this specifically so I'm curious if I'm doing something wrong.

Thanks for your help!

**Comment by LekoArts** [comment]
Please see https://github.com/gatsbyjs/gatsby/discussions/37441#discussioncomment-4656159
